backend/etl/soft_story_properties_data_handler.py

```diff
@@ -65,12 +61,17 @@ def parse_data(self, data: dict) -> list[dict]:
             geometry = feature.get("geometry", {})
             if geometry:
                 geom_longitude, geom_latitude = geometry["coordinates"]
-            addresses.append(properties.get("address"))
 
             # Search for the address in the mapbox geojson
-            mapbox_coordinates = self.mapbox_geojson_manager.get_mapbox_coordinates(
+            if not self.mapbox_geojson_manager.is_address_in_geojson(
                 properties.get("address")
-            )
+            ):
+                addresses.append(properties.get("address"))
+                mapbox_coordinates = None
+            else:
+                mapbox_coordinates = self.mapbox_geojson_manager.get_mapbox_coordinates(
```

Comment:
If the address is in the dataset, retrieve its value. Note that this function might still return None, if the address can't be resolved by Mapbox. In our 1000 soft story sample, this happens for 56 addresses.